--- old/usr/lib/python3/dist-packages/univention/admin/rest/__main__.py	2023-03-20 10:50:43.786213615 +0000
+++ new/usr/lib/python3/dist-packages/univention/admin/rest/__main__.py	2023-03-21 15:23:31.103906271 +0000
@@ -101,7 +101,7 @@
         # bind sockets
         socks = []
         if args.port:
-            socks.extend(bind_sockets(args.port, '127.0.0.1', reuse_port=True))
+            socks.extend(bind_sockets(args.port, '0.0.0.0', reuse_port=True))

         if args.unix_socket:
             socks.append(bind_unix_socket(args.unix_socket))
