--- old/usr/lib/python3/dist-packages/univention/admin/rest/server/__init__.py
+++ new/usr/lib/python3/dist-packages/univention/admin/rest/server/__init__.py
@@ -214,7 +214,7 @@ class Gateway(tornado.web.RequestHandler):

         # start sub processes for each required locale
         try:
-            cls.start_processes(args.processes, args.port)
+            cls.start_processes(args.processes, args.debug, args.port)
         except Exception:
             cls.signal_handler_stop(signal.SIGTERM, None)
             raise
@@ -272,13 +272,13 @@ class Gateway(tornado.web.RequestHandler):
         return f'/var/run/univention-directory-manager-rest-{language}-{territory.lower()}.socket'

     @classmethod
-    def start_processes(cls, num_processes=1, start_port=9979):
+    def start_processes(cls, num_processes=1, debug_level=2, start_port=9979):
         languages = [
             language.split(':', 1)[0]
             for language in ucr.get('locale', 'de_DE.UTF-8:UTF-8 en_US.UTF-8:UTF-8').split()
         ]
         for language in languages:
-            cmd = [sys.executable, '-m', 'univention.admin.rest', '-l', language, '-c', str(num_processes)]
+            cmd = [sys.executable, '-m', 'univention.admin.rest', '-l', language, '-c', str(num_processes), '-d', str(debug_level)]
             language = language.split('.', 1)[0]

             sock = cls.get_socket_for_locale(language)
